plugin/trino-opensearch/src/main/java/io/trino/plugin/opensearch/ScanQueryPageSource.java

```diff
@@ -182,6 +196,9 @@ public Page getNextPage()
 
     public static Object getField(Map<String, Object> document, String field)
     {
+        if (document == null) {
```

Comment:
This should be handled by the caller. It's unnatural to attempt to get a field when the document reference is known to be null.

Reply:
removed